### ISSUE TITLE
Latest version not working, reverting back to 1.1.1 works

### ISSUE THREAD

**Issue by rtmgllc** [qdm12/dns#67]
Hi! I have been using your container for quite some time and I love it. I went to pull the new image down using the "latest" tag. When I try to start it in docker I get the following: 

```
2021/04/05 20:34:17 INFO using DNS address 127.0.0.1 internally                                                                                
2021/04/05 20:34:17 INFO generating Unbound configuration                                                                                      
2021/04/05 20:34:17 INFO starting unbound                                                                                                      
2021/04/05 20:34:17 INFO healthcheck server: listening on 127.0.0.1:9999                                                                       
2021/04/05 20:34:17 INFO [1617654857] unbound[16:0] notice: init module 0: validator                                                           
2021/04/05 20:34:17 INFO [1617654857] unbound[16:0] notice: init module 1: iterator                                                            
2021/04/05 20:34:17 INFO [1617654857] unbound[16:0] info: start of service (unbound 1.10.1).                                                   
2021/04/05 20:34:37 INFO unbound loop exited                                                                                                   
2021/04/05 20:34:37 WARN healthcheck server: shutting down (context canceled)                                                                  
2021/04/05 20:34:37 ERROR DNS is not working after 10 tries: lookup github.com on 127.0.0.11:53: server misbehaving                            
2021/04/05 20:34:37 INFO Shutdown successful   
```

This is with any version above 1.1.1. 

Am I missing something?

For context, this is the 1.1.1 output, all env variables are the same:

```
2021-04-05T20:35:37.508Z        INFO    using DNS address 127.0.0.1 internally                                                                                                                                                                                                                   
2021-04-05T20:35:37.509Z        INFO    healthcheck server: listening on 127.0.0.1:9999                                                                                                                                                                                                          
2021-04-05T20:35:37.509Z        INFO    generating Unbound configuration                                                                                                                                                                                                                         
2021-04-05T20:35:37.510Z        INFO    starting unbound                                                                                                                                                                                                                                         
2021-04-05T20:35:37.544Z        INFO    unbound: [1617654937] unbound[16:0] notice: init module 0: validator                                                                                                                                                                                     
2021-04-05T20:35:37.545Z        INFO    unbound: [1617654937] unbound[16:0] notice: init module 1: iterator                                                                                                                                                                                      
2021-04-05T20:35:37.649Z        INFO    unbound: [1617654937] unbound[16:0] info: start of service (unbound 1.10.1).                                                                                                                                                                             
2021-04-05T20:35:57.824Z        INFO    restarting Unbound the first time to get updated files                                                                                                                                                                                                   
2021-04-05T20:35:57.824Z        INFO    downloading DNSSEC root hints and named root                                                                                                                                                                                                             
2021-04-05T20:35:57.826Z        ERROR   Get "https://www.internic.net/domain/named.root": dial tcp: lookup www.internic.net on 127.0.0.11:53: server misbehaving, retrying in 10s                                                                                                                
2021-04-05T20:36:07.416Z        INFO    unbound: [1617654967] unbound[16:0] info: generate keytag query _ta-4a5c-4f66. NULL IN                                                                                                                                                                   
2021-04-05T20:36:07.827Z        INFO    downloading DNSSEC root hints and named root                                                                                                                                                                                                             
2021-04-05T20:36:11.742Z        INFO    downloading and building DNS block lists                                                                                                                                                                                                                 
2021-04-05T20:36:11.750Z        INFO    0 hostnames blocked overall                                                                                                                                                                                                                              
2021-04-05T20:36:11.751Z        INFO    9 IP addresses blocked overall                                                                                                                                                                                                                           
2021-04-05T20:36:11.757Z        INFO    generating Unbound configuration                                                                                                                                                                                                                         
2021-04-05T20:36:12.987Z        INFO    starting unbound                                                                                                                                                                                                                                         
2021-04-05T20:36:13.021Z        INFO    unbound: [1617654973] unbound[19:0] notice: init module 0: validator                                                                                                                                                                                     
2021-04-05T20:36:13.022Z        INFO    unbound: [1617654973] unbound[19:0] notice: init module 1: iterator                                                                                                                                                                                      
2021-04-05T20:36:13.122Z        INFO    unbound: [1617654973] unbound[19:0] info: start of service (unbound 1.10.1).                                                                                                                                                                             
2021-04-05T20:36:13.453Z        INFO    unbound: [1617654973] unbound[19:0] info: generate keytag query _ta-4a5c-4f66. NULL IN                                                                                                                                                                   
2021-04-05T20:36:13.460Z        INFO    unbound: [1617654973] unbound[19:1] info: generate keytag query _ta-4a5c-4f66. NULL IN      
```

**Comment by qdm12** [comment]
That's strange, I have the latest running on my server fine. Can you share the entire logs for the latest image please?

Worst case scenario, I'm about (today/tomorrow) to cut a v2.0.0 release with DoT and DoH written from scratch, and that should solve it for sure 😉

**Comment by rtmgllc** [comment]
I will get those for you this evening - unless of course 2.0.0 comes out before, then I will just try that first. I appreciate it!

**Comment by stumpylog** [comment]
I'm also noticing some oddities with 1.5.0.  My container  is listening on port 5053.

```
dig @127.0.0.1 -p 5053 archive.ubuntu.com

; <<>> DiG 9.16.1-Ubuntu <<>> @127.0.0.1 -p 5053 archive.ubuntu.com
; (1 server found)
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 3820
;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4096
;; QUESTION SECTION:
;archive.ubuntu.com.            IN      A

;; Query time: 63 msec
;; SERVER: 127.0.0.1#5053(127.0.0.1)
;; WHEN: Mon Apr 05 15:58:16 PDT 2021
;; MSG SIZE  rcvd: 47
```

vs

```
dig @1.1.1.1 archive.ubuntu.com

; <<>> DiG 9.16.1-Ubuntu <<>> @1.1.1.1 archive.ubuntu.com
; (1 server found)
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 24182
;; flags: qr rd ra; QUERY: 1, ANSWER: 2, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 1232
;; QUESTION SECTION:
;archive.ubuntu.com.            IN      A

;; ANSWER SECTION:
archive.ubuntu.com.     50      IN      A       91.189.88.142
archive.ubuntu.com.     50      IN      A       91.189.88.152

;; Query time: 23 msec
;; SERVER: 1.1.1.1#53(1.1.1.1)
;; WHEN: Mon Apr 05 15:58:48 PDT 2021
;; MSG SIZE  rcvd: 79
```

I saw nothing change in the logs when I queried.  I'm going to play around with more verbosity and see what shakes up.

**Comment by stumpylog** [comment]
The closest logging I could get before I needed to restore my network is here: https://gist.github.com/stumpylog/42bb37076411dc880916d344e9cb917b

Not the expert, but I see a lot of `nodata`, maybe that's something?

**Comment by qdm12** [comment]
Hey @stumpylog I created #68  with your content (copied pasted) as what you mention is different from the original issue. I'll delete the comments here to keep each issue focused. I also replied with a comment on the other issue.

**Comment by qdm12** [comment]
@rtmgllc Also how are you running the container? 🤔 Portainer, docker-compose, docker run, other?

**Comment by rtmgllc** [comment]
Docker run. Internet just came back up, will get the logs in a bit.

**Comment by qdm12** [comment]
Oh I see why. If the DNS resolution doesn't work after 10 tries (trying to resolve github.com), the program is designed to crash itself so the container gets restarted to hopefully restore DNS. As @stumpylog mentioned (and confirmed by me too), it seems v1.5.0 and latest are having issues resolving from time to time 🤔 Please subscribe to @stumpylog issue #68 as it will solve your issue as well. Maybe try v1.4.0? I'm myself testing things here and there, I'm not sure why Unbound refuses to work periodically, this problem doesn't seem related to my own code (wait until v2 and everything can be blamed on me then haha).

Anyway closing, let's focus on #68.

**Comment by qdm12** [comment]
Fixed, you can use v1.5.1 or re-pull :latest 

Thanks for creating the issue!
